controller/sharding/cache.go

```diff
@@ -59,7 +59,7 @@ func (s *ClusterSharding) IsManagedCluster(c *v1alpha1.Cluster) bool {
 		log.Warnf("The cluster %s has no assigned shard.", c.Server)
 	}
 	log.Debugf("Checking if cluster %s with clusterShard %d should be processed by shard %d", c.Server, clusterShard, s.Shard)
-	return clusterShard == s.Shard
+	return s.Shard != nil && clusterShard == int64(*s.Shard)
```

Comment:
s.Shard is an `int` and so is `clusterShard` here. So no need to change this logic.

Reply:
You're right. Reverted, we should rename this values to contain Ptr if they are. That's confusing as hell. 😅

Reply:
I hear you! 😅